source/MaterialXCore/Document.cpp

```diff
@@ -782,12 +830,170 @@ void Document::upgradeVersion()
 
         minorVersion = 37;
     }
+    if (majorVersion == desiredMajorVersion &&
+        minorVersion == desiredMinorVersion)
+    {
+        setVersionString(makeVersionString(majorVersion, minorVersion));
+        return;
+    }
 
-    if (majorVersion == MATERIALX_MAJOR_VERSION &&
-        minorVersion == MATERIALX_MINOR_VERSION)
+    if (majorVersion == 1 && minorVersion == 37)
     {
-        setVersionString(DOCUMENT_VERSION_STRING);
+        // Convert material Elements to Nodes
+        convertMaterialsToNode(true);
+        minorVersion = 38;
+    }
+
+    if (majorVersion >= MATERIALX_MAJOR_VERSION &&
+        minorVersion >= MATERIALX_MINOR_VERSION)
+    {
+        setVersionString(makeVersionString(majorVersion, minorVersion));
+    }
+}
+
+bool Document::convertMaterialsToNode(bool replaceNodes)
+{
+    bool modified = false;
+
+    vector<MaterialPtr> materials = getMaterials();
+    if (!materials.empty())
+    {
+        for (auto m : materials)
+        {
+            // See if a node of this name has already been created.
+            // Should not occur otherwise there are duplicate existing
+            // Material elements.
+            string materialName = m->getName();
```

Comment:
Can't this happen, albeit unlikely, is this a good way to handle it?

Reply:
I don't think this can happen as you can't load in duplicate node names and this is replacing the existing element.
I guess this should throw an exception.